build/cherry-pick.js

```diff
@@ -0,0 +1,146 @@
+const { execSync } = require('child_process');
+const conventionalCommitsParser = require('conventional-commits-parser');
+const chalk = require('chalk');
+const { version } = require('../package.json');
+
+const releaseType = process.argv[2];
+let ignoreCommits = process.argv[3];
+
+if (!releaseType) {
+	console.error(
+		'Must specify a release type:\n$ node build/cherry-pick.js [<releaseType> | minor | patch] [<ignoreCommits>]'
+	);
+	process.exit(1);
+}
+
+// doing a major release should just be merging develop into master,
+// no cherry-picking required
+if (!['minor', 'patch'].includes(releaseType)) {
+	console.error('Release type not supported:', releaseType);
+	process.exit(1);
+}
+
+if (ignoreCommits) {
+	// use the short hash if a long hash is provided
+	ignoreCommits = ignoreCommits.split(',').map(hash => hash.substring(0, 8));
+	console.log(chalk.yellow('Ignoring commits:'), ignoreCommits.join(','), '\n');
+} else {
+	ignoreCommits = [];
+}
+
+// don't run on master or develop branch
+const currentBranch = execSync('git branch --show-current')
+	.toString()
+	.trim();
+if (['develop', 'master'].includes(currentBranch)) {
+	console.error(
+		`Please run the script on a release branch and not the ${currentBranch} branch`
+	);
+	process.exit(1);
+}
+
+// find the version we should start pulling commits from
+// e.g. if we are currently on version 3.4.5 and need to do a minor release we pull commits starting from 3.4.0
+const [major, minor] = version.split('.').map(Number);
+
+let targetVersion = releaseType === 'patch' ? version : `${major}.${minor}.0`;
+
+// get all commits from a branch
+function getCommits(branch) {
+	const stdout = execSync(`git log ${branch || ''} --abbrev-commit`).toString();
+	const allCommits = stdout.split('commit ').filter(commit => !!commit);
+
+	// parse commits
+	const commits = [];
+	for (let i = 0; i < 30; i++) {
```

Comment:
Why is `30` hard-coded here?

Reply:
Ha, good catch. Forgot I left that in there while testing something.